src/utils/trust.legal.entity.bo.ts

```diff
@@ -0,0 +1,154 @@
+import { NextFunction, Request, Response } from 'express';
+import * as config from '../config';
+import { logger } from './logger';
+import { safeRedirect } from './http.ext';
+import { getApplicationData, setExtraData } from './application.data';
+import { getTrustByIdFromApp, saveLegalEntityBoInTrust, saveTrustInApp } from './trusts';
+import * as CommonTrustDataMapper from './trust/common.trust.data.mapper';
+import { mapLegalEntityTrusteeFromSessionToPage, mapLegalEntityToSession } from './trust/legal.entity.beneficial.owner.mapper';
+import { RoleWithinTrustType } from '../model/role.within.trust.type.model';
+import { ApplicationData } from '../model';
+import { CommonTrustData, TrustLegalEntityForm } from '../model/trust.page.model';
+import { Session } from '@companieshouse/node-session-handler';
+import { saveAndContinue } from './save.and.continue';
+import { FormattedValidationErrors, formatValidationError } from '../middleware/validation.middleware';
+import { validationResult } from 'express-validator';
+
+export const LEGAL_ENTITY_BO_TEXTS = {
+  title: 'Tell us about the legal entity',
+};
+
+type TrustLegalEntityBeneificalOwnerPageProperties = {
+  backLinkUrl: string,
+  templateName: string;
+  pageParams: {
+    title: string;
+  },
+  pageData: {
+    trustData: CommonTrustData,
+    roleWithinTrustType: typeof RoleWithinTrustType;
+  },
+  formData?: TrustLegalEntityForm,
+  errors?: FormattedValidationErrors,
+  url: string,
+};
+
+const getPageProperties = (
+  req: Request,
+  trustId: string,
+  isUpdate: boolean,
+  formData?: TrustLegalEntityForm,
+  errors?: FormattedValidationErrors,
+): TrustLegalEntityBeneificalOwnerPageProperties => {
+
+  return {
+    backLinkUrl: getBackLinkUrl(isUpdate, trustId),
+    templateName: getPageTemplate(isUpdate),
+    pageParams: {
+      title: LEGAL_ENTITY_BO_TEXTS.title,
+    },
+    pageData: {
+      trustData: CommonTrustDataMapper.mapCommonTrustDataToPage(getApplicationData(req.session), trustId),
+      roleWithinTrustType: RoleWithinTrustType
+    },
+    formData,
+    errors,
+    url: getUrl(isUpdate),
+  };
+};
+
+export const getTrustLegalEntityBo = (req: Request, res: Response, next: NextFunction, isUpdate: boolean): void => {
```

Comment:
so far we've has isRegistration when we need to be conditional on journey type.  Could switch for trusts but then these should at least all use isUpdate.

Reply:
For all the trust pages we've used isUpdate. If we were to change this instance the rest should also be changed for consistency